lib/middleware/csrfProtected.ts

```diff
@@ -0,0 +1,28 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { MiddlewareRequest, MiddlewareReturn } from "@lib/types";
+import { getCsrfToken } from "next-auth/client";
+import { logMessage } from "@lib/logger";
+
+export const csrfProtected = (allowMethods: string[]): MiddlewareRequest => {
+  return async (req: NextApiRequest, res: NextApiResponse): Promise<MiddlewareReturn> => {
+    try {
+      if (allowMethods && req.method && allowMethods.includes(req.method)) {
```

Comment:
Just a quick note that we don't need to check if 'allowedMethods' is 'falsey' because in the typescript declaration it's specified that 'allowedMethods' must be a String Array.